sdk/core/Azure.Core/src/RequestMethod.cs

```diff
@@ -5,24 +5,57 @@
 
 namespace Azure.Core
 {
+    /// <summary>
+    /// Represents HTTP methods sent as part of <see cref="Request"/>
+    /// </summary>
     public readonly struct RequestMethod : IEquatable<RequestMethod>
     {
+        /// <summary>
+        /// Gets the method.
```

Comment:
nit:  It may be helpful to specify "HTTP method" here;  given the context it is obvious, though due to ingrained OO terms, it took me a moment to understand.

The nomenclature of the code itself is likely fine, given that someone using it will be thinking in that context;  across the doc comments is what I was advocating for.